Tasks/Npm/npmtoolrunner.ts

```diff
@@ -74,4 +86,47 @@ export class NpmToolRunner extends tr.ToolRunner {
         let config = tl.execSync('npm', `config list ${this.dbg ? '-l' : ''}`, options);
         return options;
     }
+
+    private _getDebugLogPath(options?: tr.IExecSyncOptions): string {
+        // check cache
+        options.silent = true;
+        const cacheLocation = tl.execSync('npm', 'config get cache', options).stdout.trim();
+
+        const logs = tl.findMatch(path.join(cacheLocation, '_logs'), '*-debug.log');
+        if (logs && logs.length > 0) {
+            const debugLog = logs[logs.length - 1];
```

Comment:
As discussed offline, the chances of this happening are very low. An agent can only run one build at a time and we don't expect machines to be running multiple agents. Hosted agents should only be running one VSTS agent per VM.

I also changed the string to help users detect if there's a mismatch between the correct log and the one we displayed.